index.bs

```diff
@@ -0,0 +1,550 @@
+the user has not interacted with the device in at least |this|.{{[[threshold]]}}
+milliseconds, the [=user agent=] MUST [=queue a task=] to run the following
+steps:
+
+1.  Set |this|.{{userState}} to {{"idle"}}.
+1.  [=Fire an event=] named {{"change"}} using the {{Event}} interface at
+    |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="user becomes active">
+
+If |this|.{{[[state]]}} is `"started"`, |this|.{{userState}} is {{"idle"}} and
+the user interacts with the device in, the [=user agent=] MUST [=queue a task=]
+to run the following steps:
+
+1.  Set |this|.{{userState}} to {{"active"}}.
+1.  [=Fire an event=] named {{"change"}} using the {{Event}} interface at
+    |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+### {{IdleDetector/screenState}} attribute ### {#api-idledetector-screenstate}
+
+The <dfn attribute for=IdleDetector>screenState</dfn> attribute MUST be
+initialized to `null`.
+
+<div algorithm="screen locked">
+
+If |this|.{{[[state]]}} is `"started"` and the device is locked by the user, an
+automatic timeout or any other event, the [=user agent=] MUST [=queue a task=] to
+run the following steps:
+
+1.  Set |this|.{{screenState}} to {{"locked"}}.
+1.  [=Fire an event=] named {{"change"}} using the {{Event}} interface at
+    |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+<div algorithm="screen unlocked">
+
+If |this|.{{[[state]]}} is `"started"` and the device is unlocked by the user or
+for any other reason, the [=user agent=] MUST [=queue a task=] to run the
+following steps:
+
+1.  Set |this|.{{screenState}} to {{"unlocked"}}.
+1.  [=Fire an event=] named {{"change"}} using the {{Event}} interface at
+    |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+### {{IdleDetector/onchange}} attribute ### {#api-idledetector-onchange}
+
+<dfn attribute for="IdleDetector">onchange</dfn> is an <a>Event handler IDL
+attribute</a> for the {{change}} event type.
+
+### {{IdleDetector/requestPermission()}} method ### {#api-idledetector-requestpermission}
+
+<div algorithm>
+The <dfn method for=IdleDetector>requestPermission()</dfn> method, when invoked, MUST run these steps:
+
+1.  Let |result| be [=a new promise=].
+1.  Run these steps [=in parallel=].
+    1.  If the [=relevant global object=] of [=this=] does not have [=transient
+        activation=], [=reject=] |result| with {{NotAllowedError}} and abort
+        there steps.
+    1.  Let |permissionState| be the result of [=requesting permission to use=]
+        the [=powerful feature=] named `"idle-detection"`.
+    1.  Resolve |result| with |permissionState|.
+1. Return |result|.
+
+</div>
+
+### {{IdleDetector/start()}} method ### {#api-idledetector-start}
+
+<div algorithm>
+
+The <dfn method for=IdleDetector>start(|options|)</dfn> method, when invoked,
+MUST run these steps:
+
+1.  Let |result| be [=a new promise=].
+1.  If the [=active document=] is not [=allowed to use=] the [=policy-controlled
```

Comment:
Which active document? That's a property of a browsing context...

Probably you mean this's relevant global object's associated Document?

Reply:
I've changed some words. Please let me know if I've chosen the right ones.